src/loaders/compute_tools/tool_common.py

```diff
@@ -0,0 +1,458 @@
+            self._node_id,
+            self._data_ids,
+            self._source_data_dir,
+            self._source_file_ext,
+            self._allow_missing_files,
+            self._tool_data_id_from_filename,
+        )
+
+        # RUN tool in parallel with multiprocessing
+        args_list = []
+        for data_id, meta in genomes_meta.items():
+            output_dir = batch_dir / data_id
+            os.makedirs(output_dir, exist_ok=True)
+            args_list.append((data_id, meta[META_SOURCE_FILE], output_dir, self._debug))
+        self._execute(self._threads, tool_callable, args_list, start, False)
+        _create_metadata_file(genomes_meta, batch_dir)
+    
+    def run_batched(self, tool_callable: Callable[[Dict[str, Path], Path, int, bool], None]):
+        """
+        Run a tool in batched mode, where > 1 data file is processed by the tool in one
+        call. Each batch gets its own batch directory.
+
+        tool_callable - the callable for the tool that takes 4 arguments:
+            * A dictionary of the data_id to the source file path
+            * The output directory for results
+            * The number of threads to use for the batch
+            * A debug boolean
+        """
+        start = time.time()
+        num_batches = max(math.floor(self._threads / self._program_threads), 1)
+        # distribute genome ids evenly across batches
+        chunk_size = math.ceil(len(self._data_ids) / num_batches)
+        batch_input = []
+        metas = []
+        for batch_number, i in enumerate(range(0, len(self._data_ids), chunk_size)):
+            data_ids = self._data_ids[i: i + chunk_size]
+            batch_dir, meta = _prepare_tool(
+                self._work_dir,
+                batch_number,
+                self._node_id,
+                data_ids,
+                self._source_data_dir,
+                self._source_file_ext,
+                self._allow_missing_files,
+                self._tool_data_id_from_filename,
+            )
+            metas.append((meta, batch_dir))
+            ids_to_files = {d: meta[d][META_SOURCE_FILE] for d in data_ids}
+            batch_input.append((ids_to_files, batch_dir, self._program_threads, self._debug))
+        self._execute(num_batches, tool_callable, batch_input, start, True)
+        for meta in metas:
+            _create_metadata_file(*meta)
+
+    def _execute(
+        self,
+        threads: int,
+        tool_callable: Callable[..., None],
+        args: List[Tuple[Any]],
+        start: datetime.datetime,
+        total: bool,
+    ):
+        pool = multiprocessing.Pool(processes=threads)
+        pool.starmap(tool_callable, args)
+        pool.close()
+        pool.join()
+        prefix = "In total used" if total else "Used"
+        print(f"{prefix} {round((time.time() - start) / 60, 2)} minutes to "
+            + f"execute {self._tool} for {len(self._data_ids)} data units"
+        )
+
+
+def unpack_gz_file(gz_file: Path):
```

Comment:
should this be moved to micrtrait.py or maybe some common module?

Reply:
It's in `tool_common`...?

I don't think it should go in microtrait, as it may be used elsewhere in the future. That being said we can move it to `loader_helper.py` or something if and when it's needed elsewhere